content/docs/practices/the_zen.md

```diff
@@ -0,0 +1,118 @@
+---
+title: The Zen of Prometheus
+sort_rank: 9
+---
+
+# The Zen of Prometheus
+
+**The Zen of Prometheus** is beginner friendly set of core values and guidelines for instrumenting your applications and writing idiomatic alerts using Prometheus.
+This is a document that's intended to maintained by the Prometheus community. Feel free to [contribute](https://github.com/prometheus/docs).
+
+## Instrument first, ask questions later
+
+During development you will never know what questions you need to ask later. Software needs good instrumentation, it's not optional. Metrics are cheap. Use them generously.
+
+The First and the most important rule, if you have to remember only one thing remember this one. Instrument all the things!
+
+## Measure what users care
+
+Do your users care if your database servers are down? Do they care about your CPU saturation? No, they care about what they experience. They care about whether they can access the page that they have requested and their results are fresh. Think in terms of latencies and availability. Let your [SLO](https://landing.google.com/sre/sre-book/chapters/service-level-objectives/)s guide your instrumentation.
+
+[RED](https://grafana.com/blog/2018/08/02/the-red-method-how-to-instrument-your-services/), [USE](http://www.brendangregg.com/usemethod.html) and [The Four Golden Signals](https://landing.google.com/sre/sre-book/chapters/monitoring-distributed-systems/#xref_monitoring_golden-signals) are known frameworks to get you started.
+
+This doesn't necessarily mean that you should ignore causal metrics. Have them, spread them liberally.
+
+    Use causal metrics to answer why something is broken.
+
+## Labels are the new hierarchies
+
+**Labels are the new hierarchies**, yet more powerful and flexible. Labels are what make Prometheus strong. Using labels, one can group and aggregate measurements afterwards. Slice and dice using Labels. Remember [Instrument first, ask questions later](#instrument-first-ask-questions-later), provide much context as possible.
+
+However, you have to *use labels with care*. The reasons will explain in subsequent rules.
+
+## Avoid missing metrics
+
+Time series that are not present until something happens are difficult to deal with. To avoid this, export 0 (or NaN if 0 would be misleading) for any time series you know may exist in advance. You have to initialize your metrics with the zero values to prevent broken dashboards and misfiring alerts. For more detailed explanation check out [Existential issues with metrics](https://www.robustperception.io/existential-issues-with-metrics).
```

Comment:
NaN shouldn't be used like this.

You don't have to expose 0, but it's much tricker to deal with correctly if you don't.